### PR TITLE
Fix for issue #415. Alignment between obbca.yaml and openchain.yaml missing in PR #486.

### PR DESCRIPTION
**Pull request by adecaro**
The ECA enforces that a pair (enrollID,enrollPWD) can be used only once. A test has been added to check this behavior.

The PR contains also an alignment between obbca.yaml and openchain.yaml that was missing in the PR #486.
Some minor cleanups are also in.

Signed-off-by: Angelo De Caro adc@zurich.ibm.com

### CODE REVIEW

**Comment by adecaro** [comment]
Signed-off-by: Angelo De Caro adc@zurich.ibm.com

File: cmd/configtxgen/main.go [hunk 1]
```diff
@@ -101,7 +101,11 @@ func doOutputAnchorPeersUpdate(conf *genesisconfig.Profile, channelID string, ou
 		return errors.Errorf("org '%s' does not have any anchor peers defined", asOrg)
 	}
 
+	// Workaround for existing configs that don't contain these keys
```
Comment (jyellick) [inline cmd/configtxgen/main.go:104]: It's not really a workaround for existing configs.  It's a workaround for the fact that this function "guesses" what the state of your channel looks like, and, by adding this element to configtx.yaml, it now guesses that the element exists in the application channel, but it doesn't for any newly created channel.

File: discovery/support/config/testdata/configtx.yaml [hunk 2]
```diff
@@ -72,6 +72,15 @@ Organizations:
             - Host: peer0.org1.example.com
               Port: 7051
 
+        # PrivateDataImplicitCollection stores the configuration for this org's implicit
```
Comment (jyellick) [inline discovery/support/config/testdata/configtx.yaml:75]: It's done, so, I'm not willing to say there's harm in including it.  But, you'll note that this yaml isn't actually consumed/used anywhere in the integration tests, and I expect for almost every user of `configtxgen` we'll never actually want/need to specify this.  You could simplify this PR considerably by simply removing the `configtxgen` support for it.  I'll comment more generally on the approach in the overall comments.

File: gossip/service/eventer.go [hunk 3]
```diff
@@ -112,6 +114,10 @@ func (ag *appGrp) AnchorPeers() []*peer.AnchorPeer {
 	return ag.anchorPeers
 }
 
+func (ag *appGrp) PrivateDataImplicitCollection() *peer.PrivateDataImplicitCollection {
```
Comment (jyellick) [inline gossip/service/eventer.go:117]: Why is this needed? It feels to me like if gossip needs a custom wrapper for the app org data, they should be declaring their own interface and using that.  Here, we're adding a function to a local implementation of an interface to satisfy a method of that interface that is never invoked by the gossip code, it seems.
